viewing_party/party.py

```diff
@@ -1,23 +1,220 @@
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    watched_list = user_data["watched"]
+    friends_watched_list = user_data["friends"]
+    watched_list_set = set()
+    friends_watched_list_set = set()
+    unique_watched_list = []
+
+### if watched_list empty, return empty list
+    if not watched_list:
+        return []
+### if friends_watched_list empty, return user's watched_list
+    elif not friends_watched_list:
+        return watched_list
+### creating set of user's watched titles
+    for movie_dict in watched_list:
+        watched_list_set.add(movie_dict["title"])
+### creating set of user's friends' watched titles
+    for friend in friends_watched_list:
+        for movie_dict in friend["watched"]:
+            friends_watched_list_set.add(movie_dict["title"])
+### checking for titles in user watched_list that are not also present in friends_watched_list
+    unique_watched_set = watched_list_set - friends_watched_list_set
+### if complete overlap, return empty list
+    if len(unique_watched_set) == 0:
+        return []
+### else return list of dictionaries with title, genre, rating for unique items
+    for title in unique_watched_set:
+        for movie_dict in watched_list:
+            if movie_dict["title"] == title:
+                unique_watched_list.append(movie_dict)
+    
+    return unique_watched_list
+
+def get_friends_unique_watched(user_data):
+    watched_list = user_data["watched"]
+    friends_watched_list = user_data["friends"]
+    watched_list_set = set()
+    friends_watched_list_set = set()
+    friends_unique_watched_list = []
+
+### if watched_list empty, return empty list
+    if not friends_watched_list:
+        return []
+### creating set of user's watched titles [same as prior]
+    for movie_dict in watched_list:
+        watched_list_set.add(movie_dict["title"])
+### creating set of user's friends' watched titles [same as prior]
+    for friend in friends_watched_list:
+        for movie_dict in friend["watched"]:
+            friends_watched_list_set.add(movie_dict["title"])
+### checking for titles in user watched_list that are not also present in friends_watched_list
+    friends_unique_watched_set = friends_watched_list_set - watched_list_set
+### if complete overlap, return empty list
+    if len(friends_unique_watched_set) == 0:
+        return []
+### else return list of dictionaries with title, genre, rating for unique items
+    for title in friends_unique_watched_set:
+        for friend in friends_watched_list:
+            for movie_dict in friend["watched"]:
+                if movie_dict["title"] == title:
+                    # for entry in friends_unique_watched_list:
+                        if movie_dict not in friends_unique_watched_list:
+                            friends_unique_watched_list.append(movie_dict)
+                    
+    return friends_unique_watched_list

```

Comment:
So, the general feedback for this function is that you might have wanted to create a helper function help make the code a bit more concise and tangible. The function could have done something like get the movies that the friends have watched, and then you could just loop through the user's watched movies to see if their respective movies were in the in entire list of friends movies.